DockerReadme.md

```diff
@@ -0,0 +1,70 @@
+## Configuring the docker environment.
+
+## For Linux you can install docker directly, but for mac and windows a linux vm is required to host the docker containers.
+
+Installation instructions for mac and windows can be found;
+
+here
+
+https://docs.docker.com/installation/mac/
+
+and here
+
+https://docs.docker.com/installation/windows/
+
+## Building the docker container. (This assumes boot2docker is running and working as per the instructions above if you are
+on mac or windows.)
+
+ # background
+Docker requires a container to be built in order for it to run our code. Our current scenario downloads a base image
+that is listed in the Dockerfile. That image contains installations of node, gulp and bower. Once that images is
+downloaded it then runs apt-get to install updates to the packages so we are using the latest versions. (This may not be
+the desired behavior at some point, but during development it is how things are working.)
+
+Once the container has been updated to use the newest versions of the software we need, it will run through the build
+process using npm install and gulp, and when that is done the process is complete.
+
+Note: This seems to have issues on slower network connections, and can take a few minutes to happen.
+
+ # build
+run the following command from a
+docker build -t="atsid/permissive" .
+
+## Setting up the environment variables.
+
+At this time the permissive configuration values will be set with an environment file when running from docker. In order
+to do this a file with the following environment variables is needed.
+
+GITHUB_ORG=<org name of repos to be managed.>
+GITHUB_TOKEN=<auth token created by an admin user of the org. needed to have permission to edit permissions.>
+GITHUB_CLIENTID=<github app client id, required for github authentication.>
+GITHUB_CLIENT_KEY=<github app client secret, required for githb authentication.>
+HOSTNAME=<host name of the app, this must match the host named in the githup client app.>
+
+## Running the docker container
+
+enter the following command to run the container in the background. (no logging will show up on the console this way.)
+docker run -d -p <host port>:<container port> --env-file <path> atsid/permissive
+
+enter this command if you would prefer to see logging on the console.
+docker run -p <host port>:<container port> --env-file <path> atsid/permissive
+
+Break this down
+
+docker run instructs docker to start an image.
+the -d flag set's it to run in the background.
+the -p flag and the arguments set it so the host will allow the ports to pass through to the ports the app has exposed. permissive
+uses port 3000, and that is what is exposed in our Dockerfile used for build so for our purposes it would be -p 3000:3000
+in order to expose port 3000 on the docker container host and the docker container itself.
+the --env-file <path> takes the path of the environment variables file.
+atsid/permissive is the docker container to run.
```

Comment:
Add note that the app is now running and ready to interact with - you need to use boot2docker ip (described below) to find URL for browser.